compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -736,7 +737,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     fn next_id(&mut self) -> hir::HirId {
         let owner = self.current_hir_id_owner;
         let local_id = self.item_local_id_counter;
-        assert_ne!(local_id, hir::ItemLocalId::new(0));
+        debug_assert_ne!(local_id, hir::ItemLocalId::new(0));
```

Comment:
Not very hot (only called for nodes created by syntactic sugar), and the assert is trivial.
Not sure about usefulness of this assert.